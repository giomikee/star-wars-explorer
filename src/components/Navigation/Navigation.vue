<template>
  <v-container class="d-flex justify-center">
    <div class="d-flex flex-md-row flex-column ga-2">
      <v-btn
        aria-label="Search for planets"
        prepend-icon="mdi-earth"
        size="large"
        @click="navigate(PAGES.planets)"
      >
        Search for planets
      </v-btn>
      <v-btn
        aria-label="Search for people"
        prepend-icon="mdi-account-multiple"
        size="large"
        @click="navigate(PAGES.people)"
      >
        Search for people
      </v-btn>
    </div>
  </v-container>
</template>

<script setup lang="ts">
  import { PAGES } from '../../constants'
  import { useNavigation } from './composables/useNavigation'

  const { navigate } = useNavigation()
</script>

<script lang="ts">
  export default { name: 'Navigation' }
</script>
