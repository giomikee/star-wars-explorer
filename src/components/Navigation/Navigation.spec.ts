import { shallowMount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import Navigation from './Navigation.vue'

const createWrapper = () => shallowMount(Navigation)

describe('Given Navigation component', () => {
  describe('When component is rendered', () => {
    it('Then it should match component snapshot', () => {
      const { element } = createWrapper()

      expect(element).toMatchSnapshot()
    })
  })
})
