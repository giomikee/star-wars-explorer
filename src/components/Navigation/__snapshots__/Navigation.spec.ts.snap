// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Given Navigation component > When component is rendered > Then it should match component snapshot 1`] = `
<div
  class="v-container v-locale--is-ltr d-flex justify-center"
>
  <div
    class="d-flex flex-md-row flex-column ga-2"
  >
    <v-btn-stub
      aria-label="Search for planets"
      block="false"
      border="false"
      density="default"
      disabled="false"
      exact="false"
      flat="false"
      icon="false"
      loading="false"
      prependicon="mdi-earth"
      replace="false"
      ripple="true"
      size="large"
      slim="false"
      stacked="false"
      symbol="Symbol(vuetify:v-btn-toggle)"
      tag="button"
      tile="false"
      variant="elevated"
    />
    <v-btn-stub
      aria-label="Search for people"
      block="false"
      border="false"
      density="default"
      disabled="false"
      exact="false"
      flat="false"
      icon="false"
      loading="false"
      prependicon="mdi-account-multiple"
      replace="false"
      ripple="true"
      size="large"
      slim="false"
      stacked="false"
      symbol="Symbol(vuetify:v-btn-toggle)"
      tag="button"
      tile="false"
      variant="elevated"
    />
  </div>
</div>
`;
